plugin-infra/go-plugin-access/test/com/thoughtworks/go/plugin/access/pluggabletask/PluggableTaskPluginInfoBuilderTest.java

```diff
@@ -0,0 +1,104 @@
+import com.thoughtworks.go.plugin.infra.Action;
+import com.thoughtworks.go.plugin.infra.plugininfo.GoPluginDescriptor;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+public class PluggableTaskPluginInfoBuilderTest {
+    private TaskExtension extension;
+
+    @Before
+    public void setUp() throws Exception {
+        extension = mock(TaskExtension.class);
+
+        TaskConfig taskConfig = new TaskConfig();
+        taskConfig.add(new TaskConfigProperty("username", null).with(Property.REQUIRED, true).with(Property.SECURE, false));
+        taskConfig.add(new TaskConfigProperty("password", null).with(Property.REQUIRED, true).with(Property.SECURE, true));
+
+        TaskView taskView = new TaskView() {
+
+            @Override
+            public String displayValue() {
+                return "my task plugin";
+            }
+
+            @Override
+            public String template() {
+                return "some html";
+            }
+        };
+
+
+        final Task task = mock(Task.class);
+        when(task.config()).thenReturn(taskConfig);
+        when(task.view()).thenReturn(taskView);
+
+        final GoPluginDescriptor descriptor = mock(GoPluginDescriptor.class);
+        String pluginId = "plugin1";
+        when(descriptor.id()).thenReturn(pluginId);
+
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                final Action<Task> action = (Action<Task>) invocation.getArguments()[1];
+                action.execute(task, descriptor);
+                return null;
+            }
+        }).when(extension).doOnTask(eq("plugin1"), any(Action.class));
+    }
+
+    @Test
+    public void shouldBuildPluginInfo() throws Exception {
+        GoPluginDescriptor descriptor = new GoPluginDescriptor("plugin1", null, null, null, null, false);
+
+        PluggableTaskPluginInfo pluginInfo = new PluggableTaskPluginInfoBuilder(extension) .pluginInfoFor(descriptor);
+
+        List<PluginConfiguration> pluginConfigurations = Arrays.asList(
+                new PluginConfiguration("username", new Metadata(true, false)),
+                new PluginConfiguration("password", new Metadata(true, true))
+        );
+        PluginView pluginView = new PluginView("some html");
+
+        assertThat(pluginInfo.getDescriptor(), is(descriptor));
+        assertThat(pluginInfo.getExtensionName(), is("task"));
+        assertThat(pluginInfo.getDisplayName(), is("my task plugin"));
+        assertThat(pluginInfo.getTaskSettings(), is(new PluggableInstanceSettings(pluginConfigurations, pluginView)));
+    }
+}
```

Comment:
That is not possible, because if the plugin does not provide a task or settings, the `JsonBasedTaskExtensionHandler_V1` will blow up appropriately.